juju/juju
# Raft API Client

~~**Requires https://github.com/juju/juju/pull/13342 to land first**~~

The following adds a raftlease client. The aim is to provide a robust
way of sending requests whilst being honest about the error messages
that fail. In particular, we drop requested commands when we don't have a
connection and it's up to the callee to retry again. Previous
implementations just dropped them on the floor.

## QA steps

Current tests pass. As this hasn't been turned on yet, we just need
to ensure that existing implementations work.

```sh
$ juju bootstrap lxd test
$ juju model-config -m controller logging-config="<root>=INFO;juju.worker.lease.raft=TRACE;juju.core.raftlease=TRACE;juju.worker.globalclockupdater.raft=TRACE;juju.worker.raft=TRACE"
$ juju enable-ha
$ juju debug-log -m controller
```
